delta-io/delta-kernel-rs
# Resolve deletion vectors to find inserted and removed rows for CDF

## What changes are proposed in this pull request?
This adds a function `resolve_scan_file_dv` which converts `CdfScanFile` to `ResolvedCdfScanFile` by reading deletion vectors, and reconciling paired add/remove deletion vectors.

`ResolvedCdfScanFile` is a struct that holds a `CdfScanFile` with its selection vector. It is guaranteed that the `scan_type` of `CdfScanFile` will match the `_change_type` of all the selected rows in its `path` if it is type add or remove.

`treemap_to_bools` is split into `deletion_treemap_to_bools` and `selection_treemap_to_bools`, which generate selection vectors from either a deletion treemap or selection treemap respectively.

`get_selection_vector` now depends on a new helper function `get_treemap`.

## How was this change tested?
The following are tested:
- adding a deletion vector to a file that did not have one
- Removing a deletion vector from a file that had one
- restoring a subset of deleted rows
- deleting a subset of remaining rows after a deletion already happened
- Deleting some rows and restoring others
- Remove_dv on type cdc and remove fails
- Add, cdc, and rm with no deletion vector return a `None` selection vector.
- ensure that the correct selection vector is generated for `selection_treemap_to_bool`
